in_toto/slsa_provenance/v1.0/provenance.go

```diff
@@ -86,29 +86,39 @@ type ProvenanaceRunDetails struct {
 	// In most cases, this SHOULD NOT contain all intermediate files generated
 	// during the build. Instead, this SHOULD only contain files that are
 	// likely to be useful later and that cannot be easily reproduced.
-	Byproducts []ArtifactReference `json:"byproducts,omitempty"`
+	Byproducts []ResourceDescriptor `json:"byproducts,omitempty"`
 }
 
-// ArtifactReference describes a particular artifact. At least one of URI or
-// digest MUST be specified.
-type ArtifactReference struct {
-	// URI describing where this artifact came from. When possible, this SHOULD
-	// be a universal and stable identifier, such as a source location or
-	// Package URL (purl).
+// ResourceDescriptor describes a particular software artifact or resource
+// (mutable or immutable).
+// See https://github.com/in-toto/attestation/blob/main/spec/v1.0/resource_descriptor.md
+type ResourceDescriptor struct {
```

Comment:
Thoughts on using ResourceDescriptor from https://github.com/in-toto/attestation/blob/main/go/v1/resource_descriptor.pb.go?

cc @marcelamelara

Reply:
I think that's a good idea. But given the proto only defines ResourceDescriptor and misses many other fields of SLSA v1 predicate, should we move on with current approach and cut a release first, and later switch over to models generated from proto once all v1 predicate fields are well defined in attestation repo?

Reply:
To my understanding, SLSA already defines full protos for SLSA provenance v1: https://github.com/slsa-framework/slsa/tree/main/docs/provenance/schema/v1

We (in-toto attestation maintainers) were operating under the assumption that SLSA will maintain all SLSA-related predicate spec and proto versions. Based on your comment here, it sounds like there's an expectation that the in-toto attestation repo should maintain (a separate copy) of SLSA provenance v1 protos?

Reply:
Oh, I was only looking at https://github.com/in-toto/attestation/tree/main/protos/in_toto_attestation/v1. I thought intoto will define its own proto again.

If intoto can reuse the proto from SLSA, that would be great. I have no option on this :D I'd defer intoto maintainers to decide this.

I am looking forward to seeing an intoto-golang release for SLSA v1!

Reply:
Got it :) At least my preference would be to use the SLSA-defined protos for generating the language bindings for the predicate, so this integration is something we should definitely discuss on our end.

Reply:
Sounds good to me! Is there a target date we can expect a release with the slsa v1 support? Thanks!

Reply:
I think I'm okay to merge this PR and then next into main and cut a release to unblock other tools. My inclination is to switch it out behind the scenes later as we want to do that for multiple models, not just SLSA Provenance.

Reply:
+1 with @adityasaky can merge for now and switch behind the scenes in another release.